Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs

```diff
@@ -44,10 +55,63 @@ static HttpAuthorizationRequestHeaderValidator()
             {
                 Host = host,
                 Key = key,
-                Resource = groups.ContainsKey("resource") ? groups["resource"] : string.Empty,
+                Resource = (groups.ContainsKey("resource") && groups["resource"] != "/")
+                    ? groups["resource"]
+                    : string.Empty,
             }.ToString();
 
             return nameof(ValidationState.Unknown);
         }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message)
+        {
+            var fingerprint = new Fingerprint(fingerprintText);
+            string host = fingerprint.Host;
+
+            try
+            {
+                using HttpClient client = CreateHttpClient();
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Guid.NewGuid().ToString());
+
+                using HttpResponseMessage responseDummy = client
+                    .GetAsync(host + (fingerprint.Resource ?? string.Empty), HttpCompletionOption.ResponseHeadersRead)
```

Comment:
>fingerprint.Resource [](start = 38, length = 20)

please lift this into its own variable. generally, it's helpful to show early in the validator what data from the fingerprint is used in the method. this approach also helps lead to more concise lines of code, as a variable named 'resource' is fewer columns than 'fingerprint.Resource'. Note that using a variable will allow you to handle the conditional assignment to string.empty as well, also shortening code. :)

Reply:
changed the code. Let me know what do you think :)

---
In reply to: [577118840](https://github.com/microsoft/sarif-pattern-matcher/pull/220#discussion_r577118840) [](ancestors = 577118840)